qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -149,60 +60,104 @@ def settings(self) -> Dict[str, Any]:
             "callback": self.callback,
         }
 
-    def minimize(
+    def ask(self) -> AskObject:
+        """
+        This method is the part of the interface of the optimizer that asks the
+        user/quantum_circuit how and where the function to optimize needs to be evaluated. It is the
+        first method inside of a "step" in the optimization process.
+        For gradient descent this method simply returns an AskObject containing the current point for
+        the gradient to be evaluated, a parameter epsilon in case the gradient has to be approximated
+        and a bool flag telling the prefered way of evaluating the gradient.
+        """
+        return AskObject(
+            x_jac=self._state.x,
+        )
+
+    def tell(self, ask_object: AskObject, tell_object: TellObject) -> None:
+        """
+        This method is the part of the interface of the optimizer that tells the
+        user/quantum_circuit what is the next point that minimizes the function (with respect to last
+        step). In this case it is not going to return anything since instead it is just going to update
+        state of the optimizer.It is the last method called inside of a "step" in the optimization process.
+        For gradient descent this method updates self._state.x by an ammount proportional to the learning rate
+        and the gradient at that point.
+        """
+        update = tell_object.eval_jac
+        self._state.x -= next(self._state.eta) * update
+        self._state.stepsize = np.linalg.norm(update)
+        self._state.nit += 1
+
+    def evaluate(self, ask_object: AskObject) -> TellObject:
+        """
+        This is the default way of evaluating the function given the request by self.ask().
+        For gradient descent we are going to check how to evaluate the gradient, evaluate and
+        return a TellObject.
+        """
+        if self._state.jac is None:
+            grad = Optimizer.gradient_num_diff(
+                x_center=ask_object.x_fun,
+                f=self._state.fun,
+                epsilon=self.perturbation,
+                max_evals_grouped=1,  # Here there was some extra logic I am just neglecting for now.
```

Comment:
Is this comment still open?

Reply:
Yes, this value is not 1 in the old gradient descent class, but I haven't had time to look what it makes reference to.